### DISCUSSION TITLE
Apply node-level MSE on batches of multiple graphs

### DISCUSSION THREAD

**Discussion by bardia-esmaeili** [pyg-team/pytorch_geometric#4678] [Q&A] [answered]
Hello,

I am trying to measure the below loss function for a batch of multiple graphs with a different number of nodes for each graph.

![image](https://user-images.githubusercontent.com/29665874/169215332-0aa695fa-789c-4210-a44c-f2edda81ee5b.png)

As you can see in the formula, I should measure the mse for each node embedding in each graph, measure the mean over that graph, then average the value for all graphs. 

Since a data loader in PyG provides `x, edge_index, y, batch, ptr`, is there a way to calculate such a loss over a batch in a parallel form?

**Comment by rusty1s** [comment, marked as answer]
You can use `scatter_mean` to compute the mean per graph, and then use `mean()` to compute the mean per batch.

**Reply by bardia-esmaeili** [reply]
Thank you for your fast response.
